### PR TITLE
refactor: add tx context for dealing with tx mutations

### PR DESCRIPTION
**Pull request by sborrazas**

### CODE REVIEW

File: test/integration/ae_mdw/sync/oracle_test.exs [hunk 1]
```diff
@@ -17,14 +19,16 @@ defmodule Integration.AeMdw.Db.Sync.OracleTest do
...
-               )
+        mutation =
+          OracleExtendMutation.new(
+            {Sync.height(:top), 0},
+            Util.last_txi(),
+            pubkey,
+            123
+          )
+
+        Mnesia.transaction([mutation])
```
Comment (jyeshe) [inline test/integration/ae_mdw/sync/oracle_test.exs:31]: some assertion needed

File: test/integration/ae_mdw/sync/oracle_test.exs [hunk 2]
```diff
@@ -34,14 +38,16 @@ defmodule Integration.AeMdw.Db.Sync.OracleTest do
...
```
Comment (jyeshe) [inline test/integration/ae_mdw/sync/oracle_test.exs:50]: some assertion needed

File: test/integration/ae_mdw/sync/oracle_test.exs (hunk 1 above)
Comment (sborrazas) [reply test/integration/ae_mdw/sync/oracle_test.exs:33]: Done

File: test/integration/ae_mdw/sync/oracle_test.exs (hunk 2 above)
Comment (sborrazas) [reply test/integration/ae_mdw/sync/oracle_test.exs:55]: Done

**Review by jyeshe — APPROVED** [review]

**Review by thepiwo — APPROVED** [review]
looks good as far as I understand

**Pull request merged**